### DISCUSSION TITLE
brew install nginx and nginx default configuration file missing

### DISCUSSION THREAD

**Discussion by FubinSama** [Homebrew/discussions#4520] [Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.0.18
ORIGIN: https://github.com/Homebrew/brew
HEAD: b0dc84b117f54f6b18e0528b95477c54e409ebdd
Last commit: 8 days ago
Core tap origin: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 520a92537ebf8d764e7971c2d3634df6a04dd2b3
Core tap last commit: 2 hours ago
Core tap branch: master
Core tap JSON: 21 May 04:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 14.0.0 build 1400
Git: 2.40.1 => /opt/homebrew/bin/git
Curl: 7.86.0 => /usr/bin/curl
macOS: 13.2.1-arm64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: A newer Command Line Tools release is available.
Update them from Software Update in System Settings.

If that doesn't show you any updates, run:
  sudo rm -rf /Library/Developer/CommandLineTools
  sudo xcode-select --install

Alternatively, manually download them from:
  https://developer.apple.com/download/all/.
You should download the Command Line Tools for Xcode 14.3.


Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libwkhtmltox.0.12.6.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/wkhtmltox/image.h
  /usr/local/include/wkhtmltox/pdf.h

Warning: You have an unnecessary local Cask tap.
This can cause problems installing up-to-date casks.
Please remove it by running:
  brew untap homebrew/cask

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core

Warning: Your Xcode (14.2) is outdated.
Please update to Xcode 14.3 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

I used `brew install nginx` to install the nginx and don't have any error.
The `brew info nginx` said: `The default port has been set in /opt/homebrew/etc/nginx/nginx.conf to 8080 so that
nginx can run without sudo.`
But I don't have the `/opt/homebrew/etc/nginx` directory.
It seems that only the `nginx` binary file is downloaded. The relevant default configurations are missing and The runtime files (such as /opt/homebrew/var/log/nginx/error.log) are not created.

### Relevant casks

nginx: stable 1.23.4 (bottled), HEAD

**Comment by SMillerDev** [comment]
> But I don't have the /opt/homebrew/etc/nginx directory.
It seems that only the nginx binary file is downloaded. 

I just tried for myself but it was definitely installed for me. Maybe there is some virus scanner interfering?

> The runtime files (such as /opt/homebrew/var/log/nginx/error.log) are not created.

Those will be created when you actually run it with settings that create them.

**Comment by FubinSama** [comment]
@SMillerDev Thanks for your help.
> I just tried for myself but it was definitely installed for me. Maybe there is some virus scanner interfering?

It seems that  this warning (`Warning: The post-install step did not complete successfully`) is causing the problem, but I don't know how to fix it

> Those will be created when you actually run it with settings that create them.

I need to create the directory `/opt/homebrew/var/log/nginx`, so that the `error.log` can create.

**Reply by SMillerDev** [reply]
> Warning: The post-install step did not complete successfully

Try `brew postinstall --verbose nginx` for more detail

**Comment by FubinSama** [comment]
@SMillerDev Thanks very much! Because I set `export USER='xxx'` in my ~/.zsh_rc for my program auto runing as the dev mode.  But The homebrew also read this env variables and run error.

**Reply by SMillerDev** [reply]
That will probably mess up a lot tools that assume `$USER` is set by the OS to the username.
